src/microbe_stage/editor/HexPopupMenu.cs

```diff
@@ -175,6 +177,32 @@ protected override void ResolveNodeReferences()
 
     protected abstract void UpdateMoveButton();
 
+    protected override Vector2 CalculateSize()
+    {
+        var size = base.CalculateSize();
+        if (titleLabel != null && normalFont != null)
+        {
+            // reset size of label to update it
+            // otherwise rect height doesn't update to new value soon enough
+            titleLabel.SetSize(Vector2.Zero);
+
+            size = new Vector2(size.x,
+                size.y +
+                Mathf.Abs(
+                    (titleLabel.GetVisibleLineCount() *
+                        normalFont.GetHeight()) -
+                    titleLabel.RectSize.y));
+
+            // make the container visible because for whatever
+            // reason resetting size of text makes it invisible
+            // I don't know why this works, but it just works
+            container.Visible = false;
+            container.Visible = true;
```

Comment:
The container probably sizes itself when it becomes visible. Or something like that. I don't really look forward to trying to port this change to Godot 4 as this seems to rely on really specific Godot under the hood stuff happening. This is why I made that comment that it'd be excellent if there was a way to let the popup naturally size itself correctly as I think that has a much lower chance of something going wrong when we can switch to Godot 4.

Reply:
Thinking about it more, I think I made this change because the changing of size that was there before broke the moment the minimum size wasn't zero, which it wasn't in earlier versions where I experimented with changing minimum size. So can probably be safely reverted presuming the CustomPopupMenu insures the minimum size is always zero.

Reply:
So, still not entirely sure why this works, but better understand what it is doing. Apparently resetting the size gives the vbox container's content absurd margins that push them well out of the content clipping area. Setting visibility appears to fix the margins to where they should be. Why it does this, I don't know. My guess is it has to do with the canvas item being updated / redrawn on being made visible when it wasn't visible before.

Reply:
Hmm, only other thing I've found is others apparently experiencing quite similar issues; even in Godot 4 apparently. (Maybe it's been fixed in later versions tho?)

https://ask.godotengine.org/142597/there-make-container-panel-expand-when-adding-more-children

Reply:
Mm, wait. Maybe relevant issue comment in the Godot engine. Mentions size update relying on drawing which does fit with the problem here and inexplicable solution. So the hiding and showing of the control is probably working because it is more or less forcing a size update due to redraw? Not sure, but it is something perhaps in the vaguely right direction. https://www.github.com/godotengine/godot/issues/20623#issuecomment-1508910488

Reply:
Yeah, I think that makes sense. I was thinking along the same lines that Godot when something becomes visible performs various actions to fix it for display.